File: tests/fixtures/codegen/dimensions/pointers-expected/queries/arrays/getview.go
package arrays

import (
	"context"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
	"github.com/jackc/pgx/v5/pgtype"
)

const GetViewSQL = "SELECT matrix, flexible FROM array_view;"

type GetViewParams struct {
}
type GetViewRow struct {
	Matrix   [][]int32           `db:"matrix"`
	Flexible pgtype.Array[int32] `db:"flexible"`
}

func (q *Queries) GetView(ctx context.Context) (GetViewRow, error) {
	var row GetViewRow
	err := q.db.QueryRow(ctx, GetViewSQL).Scan(pgsidpgx.ArrayTarget[int32](&row.Matrix, []int{2}, "GetView", "matrix", false), pgsidpgx.ArrayTarget[int32](&row.Flexible, []int{1, 2}, "GetView", "flexible", false))
	return row, err
}

File: tests/fixtures/codegen/dimensions/pointers-expected/typescript/runtime/queries/arrays/GetView.ts
import type { GetViewParams, GetViewRow } from "../../../types/queries/arrays/GetView.js";
export type { GetViewParams, GetViewRow } from "../../../types/queries/arrays/GetView.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getViewSql = "SELECT matrix, flexible FROM array_view;";
export async function getView(db: pgsid.Queryable): Promise<GetViewRow | undefined> {
    const result = await db.query(getViewSql, []);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetViewRow;
}
